applications/TopologyOptimizationApplication/custom_elements/small_displacement_simp_element.cpp

```diff
@@ -0,0 +1,286 @@
+{
+    return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Create( IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties ) const
+{
+    return Kratos::make_intrusive<SmallDisplacementSIMPElement>( NewId, pGeom, pProperties );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+SmallDisplacementSIMPElement::~SmallDisplacementSIMPElement()
+{
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+Element::Pointer SmallDisplacementSIMPElement::Clone (
+    IndexType NewId,
+    NodesArrayType const& rThisNodes
+    ) const
+{
+    KRATOS_TRY
+
+    SmallDisplacementSIMPElement::Pointer p_new_elem = Kratos::make_intrusive<SmallDisplacementSIMPElement>(NewId, GetGeometry().Create(rThisNodes), pGetProperties());
+    p_new_elem->SetData(this->GetData());
+    p_new_elem->Set(Flags(*this));
+
+    // Currently selected integration methods
+    p_new_elem->SetIntegrationMethod(BaseType::mThisIntegrationMethod);
+
+    // The vector containing the constitutive laws
+    p_new_elem->SetConstitutiveLawVector(BaseType::mConstitutiveLawVector);
+
+    return p_new_elem;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+
+// =============================================================================================================================================
+// STARTING / ENDING METHODS
+// =============================================================================================================================================
+
+//************************************************************************************
+//************************************************************************************
+
+	
+void SmallDisplacementSIMPElement::CalculateOnIntegrationPoints(const Variable<double>& rVariable, std::vector<double>& rValues,
+        const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    // From original SmallDisplacementElement
+    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints(mThisIntegrationMethod);
+
+
+    if (rValues.size() != integration_points.size())
+        rValues.resize(integration_points.size(), false);
+
+    if (rVariable == VON_MISES_STRESS) {
```

Comment:
What's the difference with the base class method?

Reply:
SmallDisplacement derives from BaseSolidElement, which computes VON_MISES_STRESS